### PR TITLE
hostap: Update EAP related config options

### PR DESCRIPTION
**Pull request by MaochenWang1**
Remove the duplicate EAP config definitions. 
EAP_ALL should select all the EAP method supported.

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -234,10 +235,26 @@ config EAP_SIM
...
+	select EAP_MSCHAPV2
+	select EAP_LEAP
+	select EAP_PSK
+	select EAP_FAST
+	select EAP_PAX
+	select EAP_SAKE
+	select EAP_GPSK
+	select EAP_PWD
+	select EAP_EKE
+	select EAP_IKEV2
```
Comment (krish2718) [inline modules/hostap/Kconfig:257]: these should be limited to what is currently supported in Zephyr, like TLS/TTLS/PEAP/GTC.

**Review by krish2718 — CHANGES_REQUESTED** [review]
can you please split this in to two commits? `removing duplicates` and `extending ALL`?

File: modules/hostap/Kconfig (hunk 1 above)
Comment (MaochenWang1) [reply modules/hostap/Kconfig]: Okay, also add EAP_FAST, which will upstream soon.

**Review by jukkar — APPROVED** [review]

**Comment by MaochenWang1** [comment]
Hi @krish2718  please review

File: modules/hostap/Kconfig [hunk 2]
```diff
@@ -459,51 +462,9 @@ config WPS
...
-
-config EAP_PWD
-	bool
-
-config EAP_EKE
-	bool
-
-config EAP_IKEv2
-	bool
-
```
Comment (jerome-pouiller) [inline modules/hostap/Kconfig:506]: Note: EAP_IKEv2 and EAP_MSCHAPv2 are not duplicated. However, they are orphans.

Comment (MaochenWang1) [reply modules/hostap/Kconfig:506]: they are already in line 228 and line 204

Comment (jerome-pouiller) [reply modules/hostap/Kconfig:506]: Indeed, I got trapped by the upper/lower case difference.

**Review by jerome-pouiller — APPROVED** [review]
Verified with:
```sed -nre  's/config (.*)/\1/p' Kconfig | sort | uniq -di```

**Review by krish2718 — APPROVED** [review]

**Pull request merged**